cpp/src/arrow/compute/kernels/vector_sort_benchmark.cc

```diff
@@ -56,17 +57,99 @@ static void SortToIndicesInt64Compare(benchmark::State& state) {
   auto max = std::numeric_limits<int64_t>::max();
   auto values = rand.Int64(array_size, min, max, args.null_proportion);
 
-  SortToIndicesBenchmark(state, values);
+  ArraySortIndicesBenchmark(state, values);
 }
 
-BENCHMARK(SortToIndicesInt64Count)
+static void TableSortIndicesBenchmark(benchmark::State& state,
+                                      const std::shared_ptr<Table>& table,
+                                      const SortOptions& options) {
+  for (auto _ : state) {
+    ABORT_NOT_OK(SortIndices(*table, options).status());
+  }
+  state.SetItemsProcessed(state.iterations() * table->num_rows());
+}
+
+static void TableSortIndicesInt64Count(benchmark::State& state) {
+  RegressionArgs args(state);
+
+  const int64_t array_size = args.size / sizeof(int64_t);
+  auto rand = random::RandomArrayGenerator(kSeed);
+  auto values = rand.Int64(array_size, -100, 100, args.null_proportion);
+  std::vector<std::shared_ptr<Field>> fields = {{field("int64", int64())}};
+  auto table = Table::Make(schema(fields), {values}, array_size);
+  SortOptions options({SortKey("int64", SortOrder::ASCENDING)});
+
+  TableSortIndicesBenchmark(state, table, options);
+}
+
+static void TableSortIndicesInt64Compare(benchmark::State& state) {
+  RegressionArgs args(state);
+
+  const int64_t array_size = args.size / sizeof(int64_t);
+  auto rand = random::RandomArrayGenerator(kSeed);
+
+  auto min = std::numeric_limits<int64_t>::min();
+  auto max = std::numeric_limits<int64_t>::max();
+  auto values = rand.Int64(array_size, min, max, args.null_proportion);
+  std::vector<std::shared_ptr<Field>> fields = {{field("int64", int64())}};
+  auto table = Table::Make(schema(fields), {values}, array_size);
+  SortOptions options({SortKey("int64", SortOrder::ASCENDING)});
+
+  TableSortIndicesBenchmark(state, table, options);
+}
+
+static void TableSortIndicesInt64Int64(benchmark::State& state) {
+  RegressionArgs args(state);
+
+  const int64_t array_size = args.size / sizeof(int64_t);
+  auto rand = random::RandomArrayGenerator(kSeed);
+
+  auto min = std::numeric_limits<int64_t>::min();
+  auto max = std::numeric_limits<int64_t>::max();
```

Comment:
You will probably get almost no duplicate values with this.

Reply:
Yes.
I've added narrow range cases that have many duplicate values.